src/coordinatorapi.cpp

```diff
@@ -369,6 +369,49 @@ void CoordinatorAPI::updateCoordCkptDir(const char *dir)
   _coordinatorSocket.writeAll(dir, strlen(dir) + 1);
 }
 
+void CoordinatorAPI::updateCkptDirGlobally(const char *dir)
+{
+  JASSERT(dir != NULL);
+
+  jalib::JSocket _coordinatorSocket1 = createNewSocketToCoordinator(COORD_ANY);
+  if (!_coordinatorSocket1.isValid()) {
+    return;
+  }
+
+  DmtcpMessage msg(DMT_UPDATE_CKPT_DIR_FOR_ALL_PROCESS);
+  msg.extraBytes = strlen(dir) + 1;
+  _coordinatorSocket1 << msg;
+  _coordinatorSocket1.writeAll(dir, strlen(dir) + 1);
+
+  msg.poison();
+  _coordinatorSocket1 >> msg;
+  msg.assertValid();
+  JASSERT(msg.type == DMT_UPDATE_CKPT_DIR_FOR_ALL_PROCESS_RESULT) (msg.type);
+
+  _coordinatorSocket1.close();
+}
+
+string CoordinatorAPI::getGlobalCkptDir(void)
```

Comment:
I don't think to have any complex string manipulations for this function as this just acting as query to get the checkpoint directory name from coordinator. Although just curious do you have your own implementation of string in dmtcp namespace ? If yes what is the reason for the same ?